packages/modules/viewer-react/src/services/iModel/ComponentConnection.ts

```diff
@@ -0,0 +1,260 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { BentleyError, Logger } from "@itwin/core-bentley";
+import type {
+  ChangesetIdWithIndex,
+  IModelConnectionProps,
+  IModelRpcProps,
+  RpcNotFoundResponse,
+} from "@itwin/core-common";
+import {
+  BentleyStatus,
+  IModelError,
+  IModelReadRpcInterface,
+  RpcManager,
+  RpcOperation,
+  RpcRequest,
+  RpcRequestEvent,
+} from "@itwin/core-common";
+import {
+  FrontendLoggerCategory,
+  IModelConnection,
+  IModelRoutingContext,
+} from "@itwin/core-frontend";
+
+const loggerCategory = FrontendLoggerCategory.IModelConnection;
+
+type ComponentProps = {
+  iTwinId?: string;
+  iModelId?: string;
+  changeset?: ChangesetIdWithIndex;
+};
+
+/**
+ * A Connection to a 'Component'. Components are small models to be used within a larger iModel.
+ * They contain documents which may or may not contain a .dgn/.bim/etc
+ * This similar to a `CheckpointConnection` except:
+ * - We do not hit the iModelHub to get the latest changeset.
+ * - getRpcProps is overridden to provide "correct" props for the Component model rather than those returned from getConnectionProps.
+ *
+ * @See https://docs.bentley.com/LiveContent/web/AECOsim%20Building%20Designer%20Help-v2/en/GUID-CCDE2FF6-C63E-4876-B3AE-5F2B33EF8547.html
+ * for more about components.
+ */
+
+export class ComponentConnection extends IModelConnection {
+  public iModelProps: IModelConnectionProps;
+  private _documentId: string;
+  private _isClosed?: boolean = false;
```

Comment:
Why is this an optional value? Shouldn't it just be:

```ts
  private _isClosed = false;
```

Note that doing this would also simplify `isClosed`.